src/arr/compiler/pyret.arr

```diff
@@ -145,31 +153,38 @@ fun main(args):
               display-progress: display-progress
             })
           failures = filter(CS.is-err, result.loadables)
-          when is-link(failures):
+          if is-link(failures):
             for each(f from failures) block:
               for lists.each(e from f.errors) block:
                 print-error(tostring(e))
                 print-error("\n")
               end
-              raise("There were compilation errors")
+              _ = print("There were compilation errors\n")
+              failure-code
             end
+          else:
+            success-code
           end
         else if r.has-key("run"):
-          CLI.run(r.get-value("run"), CS.default-compile-options.{
+          _ = CLI.run(r.get-value("run"), CS.default-compile-options.{
               standalone-file: standalone-file,
               compile-module: true,
               display-progress: display-progress,
               check-all: check-all
             })
+          success-code
         else:
-          print(C.usage-info(options).join-str("\n"))
-          raise("Unknown command line options")
+          _ = print(C.usage-info(options).join-str("\n"))
+          _ = print("Unknown command line options\n")
+          failure-code
```

Comment:
> just printing here gives _less_ information, because the `raise` would collect and print out a stack trace to aid debugging.

That is true, and would be more beneficial for developers. That then leads to this question: do we want using main `pyret.arr` blob to be _developer_-friendly, or _user_-friendly? I'd argue that stack traces are not user-friendly.